runtime/vm/JVMImage.cpp

```diff
@@ -130,30 +141,33 @@ JVMImage::freeSubAllocatedMemory(void* address)
 	omrthread_monitor_exit(_jvmImageMonitor);
 }
 
-void
+bool
 JVMImage::readImageFromFile()
 {
 	OMRPORT_ACCESS_FROM_OMRPORT(getPortLibrary());
 
 	// TODO: The size will need to be very large or dynamically based on whatever is written to the dump
```

Comment:
If you structure the image such that the header (`JVMImage`) is written first followed by the rest of the data then you only need to read the header. The header will tell you the start address and size of the image. Once you know that, you reserve the vmem address space and mmap the file to that memory range.

Reply:
Ok yeah we are working on fixing the read/write. The PR for it will be coming in later.

Reply:
also if we do mmap we probably won't call allocateImageMemory during warm run (and probably no need for thread locking). Will change that up